### ISSUE TITLE
Seaport matching logic causing duplicates (T-23078)

### ISSUE THREAD

**Issue by 0xRobin** [duneanalytics/spellbook#2404]
The current seaport logic produces duplicates on polygon.
Current understanding: This happens when multiple ERC1155s offers on the same token_id are accepted in 1 trade.
Example trade: 
https://polygonscan.com/tx/0x1878f6836e3d165fd57ab76292659d337a5be24713ac8ecc73ac7e0a03bf815a
We can see these records in the base_pairs
https://dune.com/queries/1807354?d=1
![image](https://user-images.githubusercontent.com/83790096/209544007-90eaf6ec-4adf-4990-969e-5f411fc9b2fb.png)
We should match offer 1 -> cons 2, offer 2 -> cons 3,..
But instead we are matching offer 1  -> cons (2,3,4,5), offer 2 -> cons (2,3,4,5), resulting in x4 duplicates.
Matching code is here: https://github.com/duneanalytics/spellbook/blob/6abab39c370afd0da3a646584b87a0061eaef1fc/models/seaport/polygon/seaport_polygon_trades.sql#L135-L146

## Possible solutions
1. We could apply a distinct at the top level, comes with a heavy computation cost though..
2. We could deduplicate the offers (we need all the distinct considerations as they contain the correct buyer fields), we would lose the sub_idx of the offer and replace it with the sub_idx of the consideration.
3. Find a solution that only ever matches 1 offer with 1 consideration, we could do a row_number() partitioned on the join keys on both tables and match on the row_number.. This also sounds computationally expensive though.

**Comment by 0xRobin** [comment]
@sohwak, do you have an insight how we should best resolve this?
cc: @jeff-dude @soispoke @dot2dotseurat

**Comment by 0xRobin** [comment]
Looks like we also have incorrect seaport data on ethereum, not sure if the root cause of the issue is the same.
```
select * from seaport_ethereum.trades
where tx_hash ='0xa4c28ec1ffe901ddc3c4b2b6e389e80ea6e56861452d54559af5dfc90787679e'
```
![image](https://user-images.githubusercontent.com/83790096/209547740-00aad280-f64e-4519-895c-d084746141fd.png)

**Comment by 0xRobin** [comment]
Ok so ethereum data is another root issue. :confused: 
This query reduces the trades to the problematic join, I tried it on other chains but for now only Polygon has duplicate records spawning from this. https://dune.com/queries/1807232

**Comment by 0xRobin** [comment]
This PR: https://github.com/duneanalytics/spellbook/pull/2406
provides a temporary fix to deduplicate the trades but doesn't solve the underlying issue.

**Comment by augustog** [comment]
Hmm... what's the joining key that we're missing in order to join offer - consideration 1:1? Is this missing in decoded data / at the contract level? (relatedly, what do decoded data look like for these problematic cases?

**Comment by 0xRobin** [comment]
I'm not sure, we would need to dive into the contract implementation of seaport to dig up how offer/consideration matching happens.

**Comment by dot2dotseurat** [comment]
> This PR: #2406 provides a temporary fix to deduplicate the trades but doesn't solve the underlying issue.

Is there any reason we haven't merged this yet? @augustog

**Comment by sohwak** [comment]
Hi @0xRobin, Sorry for the late reply. I saw that the same token_id occurs in 1 trade on Polygon. But I think this is the case of private sales (`receiver = '0x00000000000000000000000000000000'` so, `is_private = true`)
I was reported there are many misinterpreted cases of private sales including Ethereum chain. I will also check it to resolve the root cause of it.

**Comment by 0xRobin** [comment]
Thanks @sohwak!
I'm a bit confused as to why a null receiver is considered a private sale.
I would expect the opposite where both the sender and receiver are to be specified in the order data on private sales? (making the order only fullfillable by the designated counterparty for the private sale)

**Comment by sohwak** [comment]
@0xRobin, I don't know exactly about this. When I looked at private sales, and they were all in that form. That is, two rows are created in one private sale, like the seller sells to `0x0000` and the buyer buys from `0x000000`, so for me, I understand it is like an escrow model.
But the need is, deal this as a one trade, so in order to that I combine this into one trade. I made it by referring existing cases, but recently, other forms might be emerged, causing problems with existing interpretation.
